Changes from Hard Mode: Rubies and Magic, SPMMulti, SPM-RPG-Battles, and Brobot After Some Windex

L5050 opened the pull request:
Everything was committed through git subrepo, so the commit names might not be helpful as they refer to the projects and not spm-headers
Currently it does not pass build checks but I don't know why
line 447 in spm.eu0.lst has addresses that I don't know what file they go in as I got them from John's repo where he just put them in mod.h

L5050 committed:
add default_item_use_evt

L5050 committed:
fix thunder rage

L5050 committed:
shadoo shenanigans

L5050 committed:
Update spm.eu0.lst

L5050 committed:
Update spm.jp0.lst

L5050 committed:
Update spm.us0.lst

L5050 committed:
Update spm.us2.lst

L5050 committed:
Update setup_data.h

L5050 committed:
Update npcdrv.h

L5050 committed:
Update spm.eu0.lst

L5050 committed:
movement exists now

L5050 committed:
fix some bugs and add some addresses

L5050 committed:
eeeee

L5050 committed:
final fix :)

L5050 committed:
I forgot most of what I did honestly

L5050 committed:
fix evt patch not being compatible with LBL and GOTO

L5050 committed:
tc funi

L5050 committed:
bunch of spm-headers contributions and some logic changes

L5050 committed:
tc is functional

L5050 committed:
fix gravity

L5050 committed:
hitXp documentation

L5050 committed:
hitbox size documentation

L5050 committed:
finish up npcTribes

L5050 committed:
quick name change

L5050 committed:
add annoying volt shroom function

L5050 committed:
add tree block script

L5050 committed:
update makefile to use seeky's repo

SeekyCt committed:
Changes from practice codes (#22)

Breaking:
- You may no longer use -nostdinc in mods, the compilers headers are preferred over redefining types (-nostdlib is still usable to prevent linking the actual libraries)
- currentThread renamed to OS_CURRENT_THREAD
- Some OSThread fields corrected

Other:
- common.h fixes for C in mod environments
- Correct memory.h for kr0
- Create seq_load_sub.h
- Expand OSMutex.h
- Expand OSThread.h

SeekyCt committed:
Add note about -nostdinc

SeekyCt committed:
Restructure common.h

SeekyCt committed:
Remove redundant test

USE_STL no longer exists

SeekyCt committed:
Fix test macros

SeekyCt committed:
Changes from spm-core (#23)

Breaking:
None

Other:
- ptrdiff and intptr types in common.h
- float_t/double_t and ceil in math.h
- IOS_Write
- RelSection type
- seq_titleMain type

SeekyCt committed:
Changes from decomp (#24)

Breaking:
- Create include_cpp folder (an old decision I'm probably about to revert)

Other:
- Remove the now-unused SKIP_PPCDIS
- STRIPPED macro for functions that deadstripped outside of decomp
- Correct types for decomp in common.h
- Expand fadedrv.h
- Expand spmario_snd.h
- Expand gx.h
- Expand mem.h
- Expand/correct mtx.h
- Create msl exception.h and new.h
- Create some nw4r headders

SeekyCt committed:
Remove include_cpp

SeekyCt committed:
M2C fixes

SeekyCt committed:
Remove unused USE_STL defs

SeekyCt committed:
Always use standard headers in common.h

Breaking: remove intptr_t and uintptr_t

SeekyCt committed:
Remove niche power macros

SeekyCt committed:
Option to disable namespacing

L5050 committed:
finish new boss fight

L5050 committed:
fix dimentio issue (thanks yme)

L5050 committed:
ninjoe code

L5050 committed:
more ninjoe code

L5050 committed:
quick npc change

L5050 committed:
rewrite for new rpg system

L5050 committed:
new tpl stuff

L5050 committed:
tons of new changes

L5050 committed:
holy fuck this is a lot of changes

L5050 committed:
Sammer kingdom update, several reworks to how hitstun works

L5050 committed:
add new upgrade system

L5050 committed:
new pit system

L5050 committed:
fix us0 bug and continue pit development

L5050 committed:
more upgrades

L5050 committed:
massive bug fixes and shadoo changes

L5050 committed:
dark mario changes and performance change

L5050 committed:
meow

L5050 committed:
Merge pull request #2 from L5050/dev

Merge Hard Mode documentation with SPM-RPG-Battles documentation

L5050 committed:
various changes

L5050 committed:
e

L5050 committed:
additions

L5050 committed:
changes for the update

L5050 committed:
rebalance and evt_npc change

L5050 committed:
blah I need spm-headers change

L5050 committed:
git subrepo pull (merge) spm-headers

subrepo:
  subdir:   "spm-headers"
  merged:   "6d788a5"
upstream:
  origin:   "https://github.com/L5050/spm-headers.git"
  branch:   "master"
  commit:   "115b7e9"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "c06a924"

L5050 committed:
balance changes and bugs

L5050 committed:
idk

L5050 committed:
git subrepo pull (merge) spm-headers

subrepo:
  subdir:   "spm-headers"
  merged:   "fd838a4"
upstream:
  origin:   "https://github.com/L5050/spm-headers.git"
  branch:   "master"
  commit:   "4ba39a4"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "c06a924"

L5050 committed:
the rest of doopliss fight

L5050 committed:
the rest of the doopliss fight

L5050 committed:
new stuff

L5050 committed:
changes from brobot after some windex and spm-bringle-idk-turtle

L5050 committed:
git subrepo pull (merge) spm-headers

subrepo:
  subdir:   "spm-headers"
  merged:   "5b1bbba"
upstream:
  origin:   "https://github.com/L5050/spm-headers.git"
  branch:   "master"
  commit:   "8f38635"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "c06a924"

L5050 committed:
lst additions

L5050 committed:
fixes to prepare for spm-headers pull request

L5050 committed:
Merge branch 'master' into master

L5050 committed:
fix npcdrv for failed build issue

L5050 committed:
git subrepo pull (merge) spm-headers

subrepo:
  subdir:   "spm-headers"
  merged:   "831e64d"
upstream:
  origin:   "https://github.com/L5050/spm-headers.git"
  branch:   "master"
  commit:   "c43ca0c"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "c06a924"

L5050 committed:
Merge branch 'master' into master

SeekyCt commented on include/spm/eff_nice.h:
```diff
@@ -8,6 +8,9 @@ CPP_WRAPPER(spm::eff_nice)
 
 EffEntry * effNiceEntry(double param_1, double param_2, double param_3, double param_4, int param_5);
+
+void effMainFunc(EffEntry* effEntry);
```
At a glance this looks like it could be an unclear name or in the wrong header?

SeekyCt commented on include/spm/eff_zunbaba.h:
```diff
@@ -9,4 +10,6 @@ UNKNOWN_FUNCTION(func_800c1938);
 UNKNOWN_FUNCTION(func_800c1944);
 UNKNOWN_FUNCTION(func_800c1a48);
 
+EVT_DECLARE(wracktail_defeat)
```
`_evt` suffix

SeekyCt commented on include/spm/evt_eff.h:
```diff
@@ -15,5 +15,6 @@ UNKNOWN_FUNCTION(func_800e61b0);
 UNKNOWN_FUNCTION(func_800e61e0);
 UNKNOWN_FUNCTION(func_800e6210);
 EVT_UNKNOWN_USER_FUNC(func_800e6250)
+EVT_DECLARE_USER_FUNC(evt_eff_exclamation_question, 11)
```
I think we have the name of this effect from ttyd as `hatena`

SeekyCt commented on include/spm/rel/ta3_02.h:
```diff
@@ -0,0 +1,12 @@
+
+CPP_WRAPPER(spm::ta3_02)
+
+EVT_DECLARE(tree_block_trigger)
```
ta3_02 prefix

L5050 committed:
an2_08.h documentation

L5050 committed:
fairy documentation

L5050 committed:
fixed labeling

L5050 committed:
add chunk user funcs

L5050 committed:
add ttyd rotation

L5050 committed:
changes up until the O'Chunks fight

L5050 committed:
fix build issue

L5050 committed:
add eu0 address and fix pointer innaccuracy

L5050 committed:
acdrv_acDefs

L5050 committed:
mapGetWork address

L5050 committed:
add evt_npc_flagC_onoff

L5050 committed:
"and I had to find that in 3 days in a country where it isnt even supposed to be so um I hate this court"

L5050 committed:
lst changes

L5050 committed:
icondrv changes from Yme and tplpatch

L5050 committed:
add TexObj functions

L5050 committed:
add TPLGetGXTexObjFromPalette

L5050 committed:
add gx and tpl symbols

L5050 committed:
add symbols for icondrv

L5050 committed:
port eu0 symbols to the other versions

L5050 committed:
fix spm-headers decomp build issues

L5050 committed:
fix more build issues

L5050 committed:
fix common.h to include semicolons for UNKNOWN_FUNCTION

L5050 committed:
remove unnecessary nw4r semicolons

L5050 committed:
remove unnecessary spm semicolons

L5050 committed:
remove unnecessary wii semicolons

L5050 committed:
removed more unneeded semicolons

L5050 committed:
ack

L5050 committed:
fix the rest of the decomp build issues in spm

L5050 committed:
fix the rest of the decomp build issues in nw4r

L5050 committed:
fix npcmovemode

L5050 committed:
fix dt

L5050 commented:
this should be linked to [issue 26](https://github.com/SeekyCt/spm-headers/issues/26) which I can't do but otherwise this is good to merge

shiken-yme committed:
changes from pitrando

L5050 committed:
fix bad symbol

L5050 committed:
fix bad label

L5050 committed:
fix build issue

L5050 committed:
fix more build issues

L5050 committed:
final build fix (hopefully)

L5050 committed:
final build fix (hopefully)

shiken-yme committed:
add yme's us1 lst

L5050 commented:
Okay so I decided to go and merge everything from yme's headers into this pull request so that they can finally use updated headers rather than be stuck on headers two years out of date. Main things that need to be changed before this can be merged into your repo: temp_unk and evt_dimen are files yme made because they didnt know where they should be placed and to be honest I dont know either. evt_dimen can probably be merged with npc_dimeen_l.h? but thats your choice not mine

L5050 committed:
idk

JohnP55 committed:
remove some inline asm, fix bugs with badges, lot of rpgdrv re

L5050 committed:
fix mobj bugs

L5050 committed:
fix build issues

L5050 committed:
I missed 2 semicolons

L5050 committed:
how did I miss *one* semicolon

L5050 committed:
requested by john

L5050 committed:
evt_item documentation

shiken-yme committed:
Merge spmiiwg headers into main

L5050 committed:
fix build errors

L5050 committed:
more build error fixes

L5050 committed:
mod build fixes

shiken-yme commented:
turtle

L5050 committed:
ac def documentation

JohnP55 committed:
U

L5050 committed:
alternate route changes

L5050 committed:
git subrepo pull (merge) spm-headers

subrepo:
  subdir:   "spm-headers"
  merged:   "ce08884"
upstream:
  origin:   "https://github.com/L5050/spm-headers.git"
  branch:   "master"
  commit:   "78a0b8e"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "c06a924"

L5050 committed:
added symbols from PAL

L5050 committed:
we dont need two iconmains

L5050 committed:
fix bad symbol

L5050 committed:
fix name

L5050 committed:
fix name

L5050 committed:
add darken and lighten evts

L5050 committed:
add pure heart funcs

L5050 committed:
lst changes

L5050 committed:
fix addreses that were attributed to the wrong files

L5050 committed:
git subrepo commit (merge) spm-headers

subrepo:
  subdir:   "spm-headers"
  merged:   "b238d32"
upstream:
  origin:   "https://github.com/L5050/spm-headers.git"
  branch:   "master"
  commit:   "d32375a"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "c06a924"

L5050 committed:
fix poor merging

L5050 committed:
git reset

L5050 committed:
fast forward changes to new merge

L5050 committed:
revert to original commit

L5050 committed:
fix bad merge

L5050 committed:
Merge remote-tracking branch 'origin/upstream'

L5050 committed:
fix gx merge issue

L5050 committed:
fix evt_npc, gx

L5050 committed:
fix semicolon, incorrect function positioning

L5050 committed:
fix inaccurate merge

L5050 committed:
make name consistent

L5050 committed:
why is it lowercase

L5050 commented:
ok all conflicts are resolved and this should be 100% good

L5050 committed:
fix lst

L5050 committed:
merges I missed

L5050 committed:
marioGetHitboxWidth

L5050 committed:
all addresses

L5050 committed:
(thanks yme)

L5050 committed:
various contributions from yme that I said I'd commit but I forgot to

L5050 committed:
add symbol to lst

L5050 committed:
add peachNpcDispCb

L5050 committed:
add slim

L5050 committed:
most major 2-2 changes

L5050 committed:
add symbols

SeekyCt commented:
Apparently I had old comments that got included when I approved that? Feel free to ignore lol

L5050 commented:
Thank you!
